dipy/reconst/ivim.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/python
+""" Classes and functions for fitting ivim model """
+from __future__ import division, print_function, absolute_import
+import warnings
+import functools
+import numpy as np
+from scipy.optimize import curve_fit
+
+from dipy.core.gradients import gradient_table
+from dipy.reconst.base import ReconstModel
+from dipy.reconst.dti import _min_positive_signal
+
+
+def ivim_function(bvals, S0, f, D_star, D):
+    S = S0 * (f * np.exp(-bvals * D_star) + (1 - f) * np.exp(-bvals * D))
+    return S
+
+
+class IvimModel(ReconstModel):
+    """Ivim model
+    """
+
+    def __init__(self, gtab, fit_method="NLLS", *args, **kwargs):
```

Comment:
This seems to "inherit" a lot of its structure from the DTI model. I don't think that we need a the `fit_method` or the `*args` and `**kwargs` inputs here. In particular, you can move the code that is currently in `nlls_fit` into the `fit` method of this class.